### PR TITLE
feature/143-log-out

### PR DESCRIPTION
**Pull request by AyTee001**
https://github.com/BinaryStudioAcademy/bsa-2023-leetwars/assets/108694717/212d05b9-b8c7-464a-855c-318bd47805d2

https://trello.com/c/xkLGrj5N/143-log-out

### CODE REVIEW

File: frontend/src/app/shared/components/header/header.component.sass [hunk 1]
```diff
@@ -27,6 +27,19 @@
 :host ::ng-deep .change-color
   background-color: $cornflower
 
+::ng-deep .modal-content
```
Comment (Limbo2332) [inline frontend/src/app/shared/components/header/header.component.sass]:
```suggestion
:host ::ng-deep .modal-content
```

Comment (Limbo2332) [reply frontend/src/app/shared/components/header/header.component.sass]: Also, I don't see using of modal-content class

File: frontend/src/app/shared/components/header/header.component.sass [hunk 2]
```diff
@@ -27,6 +27,19 @@
...
+    background-color: $charcoal-grey
+    .modal-header
+        color: $green
+        .btn-close
+            background-color: $green
+    .modal-body, .modal-footer
+        color: $white
+    button
+        color: $white
```
Comment (Limbo2332) [inline frontend/src/app/shared/components/header/header.component.sass]:
```suggestion
    .modal-body, .modal-footer, button
        color: $white
```

**Review by Limbo2332 — CHANGES_REQUESTED** [review]

File: frontend/src/app/shared/components/header/header.component.sass (hunk 1 above)
Comment (AyTee001) [reply frontend/src/app/shared/components/header/header.component.sass]: Adding host disables that style and modal-content is needed to control the whole modal's background

**Review by Limbo2332 — APPROVED** [review]

File: frontend/src/app/shared/components/header/header.component.ts [hunk 3]
```diff
@@ -12,9 +18,42 @@ export class HeaderComponent {
 
     user: IUser;
 
-    constructor(private authService: AuthService) {
+    constructor(
+        private authService: AuthService,
+        private modalService: NgbModal,
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/header/header.component.ts:23]: we have our shared confirmation window
please use it

File: frontend/src/app/shared/components/header/header.component.sass (hunk 1 above)
Comment (tatianahutii) [reply frontend/src/app/shared/components/header/header.component.sass]: but without :host this can affect not only one component

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/shared/components/header/header.component.sass (hunk 1 above)
Comment (AyTee001) [reply frontend/src/app/shared/components/header/header.component.sass]: I added a class to the opening modal so styles will target a particular one

File: frontend/src/app/shared/components/header/header.component.ts (hunk 3 above)
Comment (AyTee001) [reply frontend/src/app/shared/components/header/header.component.ts:23]: I used ConfirmationModalComponent:         const modalRef = this.modalService.open(ConfirmationModalComponent, { windowClass: 'log-out-modal' });

**Review by apofis96 — APPROVED** [review]

**Review by tatianahutii — APPROVED** [review]

**Pull request merged**